windows/src/desktop/kmshell/main/UfrmMain.pas

```diff
@@ -222,30 +226,24 @@ procedure TfrmMain.TntFormCreate(Sender: TObject);
 
   kmcom.AutoApply := False;
 
-  with TRegistryErrorControlled.Create do  // I2890
-  try
-    if OpenKeyReadOnly(SRegKey_KeymanDesktop_CU) and ValueExists(SRegValue_ConfigurationState)
-      then FState := ReadString(SRegValue_ConfigurationState)
-      else FState := '0';
-  finally
-    Free;
-  end;
-
   InitNonVisualComponents;
 
-  XMLRenderers.RenderTemplate := 'Keyman.xsl';
-  XMLRenderers.Add(TKeyboardListXMLRenderer.Create);
-  XMLRenderers.Add(THotkeysXMLRenderer.Create);
-  XMLRenderers.Add(TOptionsXMLRenderer.Create(Self));
-  XMLRenderers.Add(TLanguagesXMLRenderer.Create);
-  XMLRenderers.Add(TSupportXMLRenderer.Create);
-
   Icon.ReleaseHandle;
   Icon.Handle := DuplicateIcon(hInstance, Application.Icon.Handle);
 
   Keyboards_Init;
   Options_Init;
 
+  FRenderPage := 'keyman'; // TODO: rename to 'configmain'
+  FXMLRenderers := TXMLRenderers.Create;
+  FXMLRenderers.RenderTemplate := 'Keyman.xsl';
```

Comment:
Is this template still in use? I notice the files above use `FRenderPage` instead of `RenderTemplate`.

Reply:
RenderPage is used to construct the URL to the server (e.g. /page/keyman); RenderTemplate refers to the .xsl that is used to generate the content. A future iteration may eliminate the XSL transforms but not at this point.

Therefore, this is still needed here. The template is still used to render the page -- in this instance it's done client-side as explained in the comments, but elsewhere it's done in the http server code.